### DISCUSSION TITLE
Adding delay between steps on same page

### DISCUSSION THREAD

**Discussion by HamzaPooki** [hakimio/ngx-ui-tour#179] [General]
Hello, I want to ask that how can I add delay between steps on the same page. I am using ng-boostrap tour library version 11.3.

**Comment by hakimio** [comment]
Hi,
`delayBeforeStepShow` config was introduced in `v12`. I don't think there is anything like that in `v11`.

**Reply by HamzaPooki** [reply]
Actually updating this library version required updating Angular version and I cannot update it right now as it will causes multiple packages dependencies issues.
Isn't there is any custom way to add delay?

**Reply by hakimio** [reply]
Then you'll have to create your own fork of the library and add the functionality to the library yourself.

**Comment by HamzaPooki** [comment]
Okay thanks.
Can you help me in this regard - how can I apply border-radius to the highlighted element?

**Reply by hakimio** [reply]
You can use `.touranchor--is-active` CSS class to style active tour anchor.
[Docs](https://hakimio.github.io/ngx-ui-tour/ng-bootstrap/Misc#styling-active-anchor)

**Reply by HamzaPooki** [reply]
I am using it but still border-radius is not applying on the active tour anchor.
Using library version "ngx-ui-tour-ng-bootstrap": "^12.6.0"

**Reply by hakimio** [reply]
Following works for me just fine:
```css
::ng-deep .touranchor--is-active {
    border-radius: 5px;
    border: 2px solid red;
}
```

If you want to apply `border-radius` to the backdrop, it's not possible. You can set `backdropConfig.offset` instead though.

**EDIT:** `backdropConfig.offset` was introduced in `ngx-ui-tour-ng-bootstrap` `v13`.

**Reply by HamzaPooki** [reply]
<img width="693" alt="Screenshot 2024-07-02 at 3 49 40 PM" src="https://github.com/hakimio/ngx-ui-tour/assets/86037312/79c0f643-1ef7-4329-83b1-9ec9cb81773c">

I am using this styling
```
::ng-deep .touranchor--is-active {
  position: relative;
  background: white;
  z-index: 1001;
  padding: 10px;
  border-radius: 8px;
}
```

Still border-radius is not applied on the element.

**Reply by hakimio** [reply]
Which element? `.touranchor--is-active` CSS class is applied to your button (**the anchor element**), not the backdrop. Inspect your button and you should see the styles applied to it.

**Reply by HamzaPooki** [reply]
<img width="966" alt="Screenshot 2024-07-02 at 4 13 00 PM" src="https://github.com/hakimio/ngx-ui-tour/assets/86037312/03f83fd7-1eeb-48af-9104-f13cc182e308">

This is how I am using the anchor element.

**Reply by hakimio** [reply]
`touranchor--is-active` class is applied to the active tour anchor. So, it's working as intended. Backdrop box can not have border radius.

**Reply by HamzaPooki** [reply]
Ohh Ok
That can be done using backdropConfig.offset right?

**Reply by HamzaPooki** [reply]
Can you help me with this?

This is happening on the element which shows after scroll.

<img width="966" alt="Screenshot 2024-07-02 at 5 19 14 PM" src="https://github.com/hakimio/ngx-ui-tour/assets/86037312/3ada6f51-e2b7-42c3-9007-d763239e7c99">

But if I change the width of my screen just by opening inspect mode

<img width="966" alt="Screenshot 2024-07-02 at 5 19 24 PM" src="https://github.com/hakimio/ngx-ui-tour/assets/86037312/9964f2a5-9567-4762-9c29-99aa6f394342">

It get fixed
